execution_workflows/APA-Scan/workflow/Snakefile

```diff
@@ -0,0 +1,337 @@
+# Method-specific rules
+
+
+#### NOTES #####################################################################
+
+# This just runs the APA-Scan.py script, which doesn't take command-line inputs
+# and instead you provide the configuration.ini file which tells the software
+# where to look for the input files (RNA-Seq and PAS-Seq), where the
+# annotation/genome files are found and where to output the result to.
+# because you don't specify input/output from the command line and you don't
+# specify the input files, the configuration.ini file is the ONLY input you
+# need
+
+################################################################################
+
+
+
+rule run_apa_scan:
+    """Run APA-Scan
+    """
+    input:
+        "configuration.ini"
+    output:
+        os.path.join(config["out_dir"], APA_SCAN_OUTPUT)
+    threads: THREADS
+    conda: os.path.join(config["envs"], CONDA_YAML)
+    group: "APA_SCAN"
+    log:
+        os.path.join(config["local_log"], APA_SCAN_OUTPREFIX + ".run_apa_scan.execute.python_script.log")
+    shell:
+        "( python3 APA-Scan.py ) &> {log}"
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Postprocessing: obtain suitable output formats (for benchmarks)
+
+
+# This rule just converts the output .xslx from APA-Scan to a csv
+rule convert_to_tsv:
+    """
+    This tool takes the xlsx output of APA-Scan and converts it to .tsv
+    """
+    input:
+        os.path.join(config["out_dir"], APA_SCAN_OUTPUT)
+    output:
+        os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + ".genenames.tsv")
+    threads: THREADS
+    conda: os.path.join(config["envs"], CONDA_YAML)
+    group: "APA_SCAN"
+    log:
+        os.path.join(config["local_log"], APA_SCAN_OUTPREFIX + ".convert_to_tsv.postprocess_APA-Scan.log")
+    shell:
+        "( xlsx2csv -d 'tab' {input} > {output} ) &> {log} "
+
+
+
+# This takes the .csv from above, changes the gene name to ensembl gene id and
+# then saves it as a both a bed file (Format 01) and as Format 02
+# (see Execution worfklows of README on GitHub: https://github.com/iRNA-COSI/APAeval/tree/main/execution_workflows )
+rule convert_gene_ids:
+    """
+    This tool takes the .tsv format of APA-Scan and converts it to the desired formatted .bed and .tsv files
+    """
+    input:
+        infile=os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + ".genenames.tsv")
+    output:
+        out01=os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_01.bed"),
+        out03=os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_03.tsv")
+    threads: THREADS
+    conda: os.path.join(config["envs"], CONDA_YAML)
+    group: "APA_SCAN"
+    log:
+        os.path.join(config["local_log"], APA_SCAN_OUTPREFIX + ".change_gene_ids_and_save.postprocess.log")
+    shell:
+          " python " + CONVERT_ID_PYSCRIPT + \
+                   " {input.infile} {output.out01} {output.out01} "
```

Comment:
```
shell:
    """
    python {CONVERT_ID_PYSCRIPT} {input.infile} {output.out01} {output.out01}
    """
```

This is fine but above would be my preferred formatting (and more fitting to Snakemake conventions). It could probably all be one line though (so no need for triple quotes) but that's my preferences creeping